### PR TITLE
Update Woo branding for reset password pages

### PR DESCRIPTION
**Pull request by ilyasfoo**
Related to #97149

## Proposed Changes

* Add `is-woo-com-oauth` to container class for more precise selector matching
* Various fixes to CSS changes so branding update does not override core profiler flow unintentionally
* Update reset password pages styling to match branding update
* Add back header subtitle text post-reset page (discussion: tp2u3Nk9arcpeNMZcylNG4-fi-616_4910#1044740940)
  > Stay with the "Log in to your account" screen with the current explanation. I think this screen does need a title, as removing it creates inconsistencies. 

## Testing Instructions

1. Set up [woocommerce-start-dev-env](https://github.com/woocommerce/woocommerce-start-dev-env), point wordpress.com to your local and sandbox `public-api.wordpress.com`
1. In your [`development.json`](https://github.com/Automattic/wp-calypso/blob/c470b69f28f940f90221730605717d06622639c7/config/development.json#L251), set `woocommerce/rebrand-2-0` to `true`
1. Run `yarn start`

#### Reset password page

1. In an incognito window, open https://woocommerce.com/sso?next=%2F
2. Click on `Lost your password?`
3. Observe the page is similar with design

image

#### Reset password post-email page

1. Enter any email and click on `Reset my password`
2. Observe the page is similar with design (except for `Lost password` link which will have underline)

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -455,7 +455,7 @@ class Login extends Component {
 				} else if ( this.props.twoFactorEnabled ) {
 					headerText = <h3>{ translate( 'Authenticate your login' ) }</h3>;
 				} else if ( currentQuery.lostpassword_flow ) {
-					headerText = null;
+					headerText = <h3>{ translate( 'Log in to your account' ) }</h3>;
```
Comment (ilyasfoo) [inline client/blocks/login/index.jsx:458]: Meant to be outside of feature flag

File: client/blocks/login/lost-password-form.jsx [hunk 2]
```diff
@@ -129,7 +129,7 @@ const LostPasswordForm = ( {
 			onSubmit={ onSubmit }
 		>
 			<div className="login__form-userdata">
-				<FormLabel htmlFor="email">{ translate( 'Your email address' ) }</FormLabel>
+				<FormLabel htmlFor="email">{ translate( 'Your email' ) }</FormLabel>
```
Comment (ilyasfoo) [inline client/blocks/login/lost-password-form.jsx:132]: Meant to be outside of feature flag

**Review by chihsuan — APPROVED** [review]
Tested well! Thanks. 🚢

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

**Pull request merged**